Documentation/contributing/docs/docsframework.rst

```diff
@@ -0,0 +1,382 @@
+
+Customization
+-------------
+
+Additional elements of customization for Cilium's documentation and defined in
+the main repository include:
+
+- Some custom CSS; see also class ``wrapped-table`` in the related CSS file
+  ``Documentation/_static/wrapped-table.css``
+
+- A "Copy" button, including a button to copy only commands from console-code
+  blocks, implemented in ``Documentation/_static/copybutton.js`` and
+  ``Documentation/_static/copybutton.css``
+
+- Custom header and footer definitions, for example to make link to Slack
+  target available on all pages
+
+- Warning banner on older branches, telling to check out the latest version
+  (these may be handled directly in the ReadTheDocs configuration in the
+  future, see also :gh-issue:`29969`)
+
+:spelling:ignore:`Algolia` search engine
+----------------------------------------
+
+- *To be documented.*
+
+Build set up
+============
+
+.. _docs_requirements:
+
+Requirements (dependencies)
+---------------------------
+
+We have two files for requirements: one where we declare and pin the core
+dependencies for the documentation build system, and that we use to generate a
+second requirement files that includes all sub-dependencies:
+
+- The base requirements are defined in
+  ``Documentation/requirements-min/requirements.txt``.
+- By running ``make -C Documentation update-requirements``, we use this file to
+  generate ``Documentation/requirements.txt``.
+
+Dependencies defined in ``Documentation/requirements-min/requirements.txt``
+should never be updated in ``Documentation/requirements.txt`` directly.
+Instead, update the former and regenerate the latter.
+
+These requirements are used to build the ``docs-builder`` Docker image.
+
+One of the dependency is the custom theme that we use for the documentation.
+
+Docker set-up
+-------------
+
+The documentation build system relies on a Docker image, ``docs-builder``, to
+ensure that the build environment is consistent across different systems.
+Resources related to this image include ``Documentation/Dockerfile`` and the
+requirement files.
+
+Versions of this image are automatically built and published to a registry when
+the Dockerfile or the list of dependencies is updated. This is handled in CI
+workflow ``.github/workflows/build-images-docs-builder.yaml``.
+
+The deployment described in this workflow should be run twice on each PR
+updating the Dockerfile or its dependencies, to ensure that the CI picks up an
+updated image.
+
+ReadTheDocs
+-----------
+
+Cilium's documentation is hosted on ReadTheDocs. The main configuration options
+are defined in ``Documentation/.readthedocs.yaml``.
+
+Some options, however, are only configurable from the ReadTheDocs web
+interface. For example, this is the case of:
+
+- The location of the configuration file in the repository
+- Redirects
+- Triggers for deployment
+
+Custom theme
+============
+
+The online documentation uses a custom theme, based on `the ReadTheDocs theme`_
```

Comment:
Punctuation nits; is there a Sphinx reason to put the period inside the reference?
```suggestion
The online documentation uses a custom theme based on `the ReadTheDocs theme`_.
```

Reply:
Missing period is a typo.